packages/mui-codemod/src/v6.0.0/list-item-button-prop/list-item-button-prop.js

```diff
@@ -0,0 +1,115 @@
+import findComponentJSX from '../../util/findComponentJSX';
+
+/**
+ * @param {import('jscodeshift').FileInfo} file
+ * @param {import('jscodeshift').API} api
+ */
+export default function transformer(file, api, options) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  const printOptions = options.printOptions;
+
+  //Rename components that have ListItem button to ListItemButton
+  findComponentJSX(j, { root, componentName: 'ListItem' }, (elementPath) => {
+    const index = elementPath.node.openingElement.attributes.findIndex(
+      (attr) => attr.type === 'JSXAttribute' && attr.name.name === 'button',
+    );
+    if (index !== -1) {
+      if(elementPath.node.openingElement.name.name === 'ListItem') {
+        elementPath.node.openingElement.name.name = 'ListItemButton';
+      }
+      elementPath.node.openingElement.attributes.splice(index, 1);
+    }
+  });
+
+  let hasButton = false;
+  root.find(j.ObjectProperty).forEach((path) => {
```

Comment:
May I ask you to refactor to use the [`findComponentDefaultProps` util](https://github.com/mui/material-ui/blob/next/packages/mui-codemod/src/util/findComponentDefaultProps.js)?

That returns a collection of `defaultProps` paths, given a component name. You can follow this example as a guideline: https://github.com/mui/material-ui/blob/next/packages/mui-codemod/src/deprecations/utils/movePropIntoSlots.js

This will make it easier to guide and debug 😊

Reply:
Sure, I will take a look at that. That is helpful!